pkg/storage/stores.go

```diff
@@ -86,6 +87,15 @@ type SubjectAccessCapableStore interface {
 
 type WatchEventType string
 
+type Lock interface {
+	Lock() error
+	Unlock() error
+}
+
+type LockManager interface {
```

Comment:
I don't mind however it's not necessarily a broker in the way other things called broker are being used in our codebase (singleton-ish constructor), this is the interface that is a "drop in" (not really) replacement for a sync.Mutex.

Since each consumer of distributed locks should each encapsulate their own instance of LockManager it might be ambigous with how existing brokers are used.

I would image `LockBroker` interface would provision instantiate some implementation of `LockManager` and that reference would be passed to things needing to consume it.

There is be an argument that we could make that the LockManager interface should be 
```
type LockManager interface {
  Lock(key string, opts... LockOption) error
  Unlock(key string, opts.. .LockOption) error
}
```

but it is harder (and maybe more expensive) to make the implementations work this way -- but don't mind switching to that interface